src/peer_transport/listener.ts

```diff
@@ -0,0 +1,35 @@
+// import type { ConnectionManager } from '@libp2p/interface-connection-manager'
+import type { PeerId } from '@libp2p/interface-peer-id'
+import type { ListenerEvents, TransportManager, Upgrader, Listener } from '@libp2p/interface-transport'
+import { EventEmitter } from '@libp2p/interfaces/events'
+import { multiaddr, Multiaddr } from '@multiformats/multiaddr'
+
+export interface ListenerOptions {
+  peerId: PeerId
+  upgrader: Upgrader
+  transportManager: TransportManager
+}
+
+export class WebRTCPeerListener extends EventEmitter<ListenerEvents> implements Listener {
+  constructor (
+    private readonly opts: ListenerOptions
+  ) {
+    super()
+  }
+
+    private listeningAddrs: Multiaddr[] = []
+    async listen (ma: Multiaddr): Promise<void> {
```

Comment:
Do we need this? Aren't we listening already via some relay?

Reply:
Probably not, but the rationale to add this is as follows:
1. Since webrtc-direct is a protocol, we can advertise support over any transport.
2. For running on node, we can use `node-webrtc` to replace all instances of `RTCPeerConnection` (webrtc-star and webrtc-direct currently do this).

Although, an argument can be made that any listener can support webrtc-direct since it is only implemented as a protocol.